physchem_calc.py

```diff
@@ -38,6 +37,16 @@ def fused_ring_count(m):
     return len(rings)
 
 
+def count_hbd_hba_atoms(m):
+    HDonorSmarts = Chem.MolFromSmarts('[$([N;!H0;v3]),$([N;!H0;+1;v4]),$([O,S;H1;+0]),$([n;H1;+0])]')
+    HAcceptorSmarts = Chem.MolFromSmarts('[$([O,S;H1;v2]-[!$(*=[O,N,P,S])]),' +
+                                     '$([O,S;H0;v2]),$([O,S;-]),$([N;v3;!$(N-*=!@[O,N,P,S])]),' +
+                                     '$([nH0,o,s;+0])]')
```

Comment:
These are constants and their initialization each time is unnecessary.
I suggest to move them into initialization of the pool and create there global variables. Just an example https://stackoverflow.com/a/28508998/895544

Reply:
2